### PR TITLE
multiple provider support

### LINKED ISSUE #13: Support multiple Git providers
Maybe we'll just focus on Github and Gitlab first, then we can work our way through Gitea and Bitbucket later on.

Github reference: https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads
Gitlab reference: https://docs.gitlab.com/ee/user/project/integrations/webhook_events.html

some reference that parses data from multiple providers: https://github.com/go-playground/webhooks

### PR DESCRIPTION
**Pull request by aldy505**
partially closes #13

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -44,21 +43,26 @@ bot.catch(async (e, ctx) => {
...
+    return;
+  }
+
   console.log(kleur.green("Telegram bot /start triggered"));
-  
-  if (applicationState.isStarted) {
-    await ctx.reply(
-      `This bot is already running since ${applicationState.startedDate.toLocaleDateString()} To restart it, please stop it first.`
-    );
+  const repositoryUrl = ctx.message.text.slice(7);
```
Comment (elianiva) [inline src/index.ts:51]: kasih comment angka 7 ini apa (panjangnya `/start ` kan), biar ngga jadi magic number

atau kalo sok ganteng jadiin `const [_, repositoryUrl] = ctx.message.text.split(" ")`

File: src/server/github.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      switch (req.headers["content-type"]) {
+        case "application/x-www-form-urlencoded": {
+          const url = new URLSearchParams(body);
+          req.body = Object.fromEntries(url.entries());
+          break;
+        }
+        case "application/json":
+        default:
+          req.body = JSON.parse(body);
+      }
```
Comment (elianiva) [inline src/server/github.ts:26]: ini body parser nanti bakal diextract buat dipake yg lain juga ngga?

**Review by elianiva — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 3]
```diff
@@ -1,6 +1,10 @@
 import type { IncomingMessage, ServerResponse } from "http";
 import { createServer } from "http";
+import console from "console";
 import { Webhooks, createNodeMiddleware } from "@octokit/webhooks";
+import { Telegraf } from "telegraf";
```
[the remark this answers is not in the document]
Comment (aldy505) [reply src/index.ts:5]: beda PR

File: src/server/github.ts (hunk 2 above)
Comment (aldy505) [reply src/server/github.ts:26]: iya. beda PR.  ini buat scaffold.

File: src/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (elianiva) [reply src/index.ts:5]: ok

**Comment by elianiva** [comment]
masih mau lanjutin ini?

**Comment by elianiva** [comment]
kalo gamau lanjutin biar aku yg lanjutin

**Comment by aldy505** [comment]
ide yang bagus, gas

**Comment by aldy505** [comment]
seems like you have too much free time lol

File: src/index.ts [hunk 4]
```diff
@@ -1,186 +1,90 @@
...
+  ReviewEventHandler,
+  VulnerabilityEventHandler
+} from "~/infrastructure/event-handlers";
+import type { AppConfig } from "~/types";
+import { GithubServer } from "~/infrastructure/server/GithubServer";
+import { GithubWebhook } from "~/application/webhook/github";
+
+// configurations
+const configFile = await readFile(path.resolve("config.gura"), { encoding: "utf-8" });
+const config = parseGura(configFile) as AppConfig;
```
Comment (aldy505) [inline src/index.ts:24]: should also validate is there is any invalid configuration being set on the gura conf file

File: src/index.ts [hunk 5]
```diff
@@ -1,186 +1,90 @@
...
+    switch (req.headers["content-type"]) {
+      case "application/x-www-form-urlencoded": {
+        const url = new URLSearchParams(body);
+        req.body = Object.fromEntries(url.entries());
+        break;
+      }
+      case "application/json":
+      default:
+        req.body = JSON.parse(body);
+    }
```
Comment (aldy505) [inline src/index.ts:49]: should not parse anything if the `body` length is 0

File: src/infrastructure/ConsoleLogger.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  private readonly _infoPrefix = c.bold(c.green("[INFO] ➤ "));
+  private readonly _errorPrefix = c.bold(c.red("[ERROR] ➤ "));
+  private readonly _warnPrefix = c.bold(c.yellow("[WARN] ➤ "));
+
+  private getTimestamp() {
+    return c.gray(new Date().toLocaleString());
+  }
+
+  debug(text: string) {
+    // eslint-disable-next-line no-console
```
Comment (aldy505) [inline src/infrastructure/ConsoleLogger.ts:15]: import console instead of disabling eslint like this

**Pull request merged**